Fixed rendering of Clebsch-Gordan (CG) class

redeboer opened the pull request:
#### References to other Issues or PRs

- Fixes #18781
- Fixes #21001


#### Brief description of what is fixed or changed

- Fixed the order of the formula in the docstring (#18781)
- Added a link to the Particle Data Group for comparing the doctests
- Improved the LaTeX rendering so that it also renders if a `CG` is squared, for instance (#21001)


#### Release Notes

<!-- BEGIN RELEASE NOTES -->
* physics.quantum
  * Added a precedence attribute to `CG` to fix latex rendering
  * Fixed argument order of the math in the `CG` documentation
<!-- END RELEASE NOTES -->

asmeurer commented:
The LaTeX thing should be fixed by fixing the precedence in the printer. If you look at precedence.py, it looks like it needs to define a `precedence` attribute (this stuff really needs to be documented).

redeboer commented:
> The LaTeX thing should be fixed by fixing the precedence in the printer. If you look at precedence.py, it looks like it needs to define a `precedence` attribute (this stuff really needs to be documented).

Ah ok, yeah that sounds like a better solution. Perhaps it's also a solution to #19661 and #20959?

redeboer commented:
As for #18781, I want to look into this a bit further, because it seems c3a1a40 is not the correct solution.

redeboer committed:
fix: correct argument order in CG docstring

redeboer committed:
docs: improve parameter section CG class

Clarifies the difference between total angular momentum and the angular
momenta of the sub-systems.

Also adds a link to Clebsch-Gordan in PDG to verify the doctests.

redeboer committed:
fix: increase LaTeX precedence of CG class

This correctly renders powers of a CG instance

redeboer commented:
> The LaTeX thing should be fixed by fixing the precedence in the printer. If you look at precedence.py, it looks like it needs to define a `precedence` attribute (this stuff really needs to be documented).

@asmeurer is this the idea? 7fd95c3c4e

asmeurer commented:
I would put the precedence on the class itself, rather than in precedence.py, since the latex printer is also defined there. I believe you can define a `precedence = PRECEDENCE["Pow"] - 1`, although you may have to read the source code in the printer to verify that this is correct. I don't know enough about the object itself to confirm if this is the correct precedence for it.

redeboer committed:
fix: define CG precedence as class attribute

redeboer commented:
Ah yes that makes more sense --> 2f5dd663dd

Hadn't noticed `precedence()` also checks class attributes:
https://github.com/sympy/sympy/blob/c07fc4733434ddb209e91a71b6abaa2e04c80e29/sympy/printing/precedence.py#L123-L124

asmeurer commented:
Looks good. Can you add release notes entries for the changes?

redeboer commented:
> Looks good. Can you add release notes entries for the changes?

Done :ok_hand:

oscarbenjamin commented:
Looks good, thanks!
